tests/AbstractTestCase.php

```diff
@@ -41,4 +52,60 @@ public function createApplication()
 
         return $app;
     }
+
+    /**
+     * Sign in an user if it's the case.
+     *
+     * @param User|null $user
+     *
+     * @return AbstractTestCase
+     */
+    protected function signIn(User $user = null)
+    {
+        $this->user = $user ?: $this->createUser();
+
+        $this->be($this->user);
+
+        return $this;
+    }
+
+    /**
+     * Create and return a new user.
+     *
+     * @param array $properties
+     *
+     * @return User
+     */
+    protected function createUser($properties = [])
+    {
+        return factory(User::class)->create($properties);
+    }
+
+    /**
+     * Set up the needed configuration to be able to run the tests.
+     *
+     * @return AbstractTestCase
+     */
+    protected function setupConfig()
+    {
+        $env = $this->app->environment();
+        $repo = $this->app->make(Repository::class);
+        $cache = $this->app->make(Cache::class);
+        $loaded = $cache->load($env);
+
+        try {
+            if ($loaded === false) {
+                $loaded = $repo->all();
+                $cache->store($env, $loaded);
+            }
+
+            $settings = array_merge($this->app->config->get('setting'), $loaded);
+
+            $this->app->config->set('setting', $settings);
+        } catch (Exception $e) {
```

Comment:
Yeah, we wouldn't want to fall on our ass in the non-testing environment but I feel like it's something I'd like to have flagged up if there is an issue when running tests?